testsuite/classlibrary/TestNetAddr.sc

```diff
@@ -53,29 +25,20 @@ TestNetAddr : UnitTest {
 			*testMsg.isArray.if({ testMsg }, { [testMsg] })
 		);
 
-		{
-			if (messageReceived.test.not) {
-				messageValue = \timeout;
-				messageReceived.unhang;
-			}
-		}.defer(1);
-
-		messageReceived.wait();
+		messageReceived.waitFor(1) { messageValue.notNil };
 
 		thisProcess.perform(unregisterMethod, func);
 
-		if (messageValue == \timeout) {
-			this.assert(false, "addr.sendRaw was never received");
-		} {
-			#msg, time, replyAddr, recvPort = messageValue;
+		if (messageValue.isNil) {
+			this.assert(false, "addr.% was never received".format(sendMethod));
```

Comment:
optional, replace if statement with: 
```
this.assert(messageValue.notNil, ...
```

Reply:
thanks a lot! however I would prefer to keep the if, so that the assert doesn't run if messageValue.notNil. I also thought a failed assert would stop the test, but since it doesn't, I've put the other tests in the else block. This way, we check fields only if message was received.